travel-event-host-app/src/components/searchSection/searchEventsFilterBox/filterBox.tsx

```diff
@@ -1,24 +1,26 @@
 'use client';
-import { useEffect, useState } from 'react';
+import CategoriesType from '@/components/searchSection/types';
+import Category from '@/lib/category';
+import CategoryDict from '@/lib/categoryArray';
+import CloseIcon from '@mui/icons-material/Close';
 import {
   Box,
-  MenuItem,
-  Select,
   Checkbox,
-  FormGroup,
   FormControlLabel,
-  Typography,
+  FormGroup,
   IconButton,
+  MenuItem,
+  Select,
+  Typography,
 } from '@mui/material';
-import categoriesArr from '@/lib/categoryArray';
-import CategoriesType from '@/components/searchSection/types';
+import { useEffect, useState } from 'react';
 import styles from '../styles.module.css';
-import CloseIcon from '@mui/icons-material/Close';
 
 const updateCategories = () => {
-  let updateCategories: { [key: string]: boolean } = {};
+  let updateCategories: { [key in string]: boolean } = {};
+
+  Object.values(Category).map((category: string) => (updateCategories[category] = false));
```

Comment:
the object that keeps track of the checkbox state now uses the `Category` enum (camelCase) as the key, instead of a string with spaces.